sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryUtils.java

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.tracing.opentelemetry;
+
+import com.azure.core.util.logging.ClientLogger;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.azure.core.util.tracing.Tracer.ENTITY_PATH_KEY;
+import static com.azure.core.util.tracing.Tracer.HOST_NAME_KEY;
+
+class OpenTelemetryUtils {
+    private static final ClientLogger LOGGER = new ClientLogger(OpenTelemetryUtils.class);
+
+    private static final Map<String, String> ATTRIBUTE_MAPPING_V1_12_0 = getMappingsV1200();
+    static final String SERVICE_REQUEST_ID_ATTRIBUTE = "serviceRequestId";
+    static final String CLIENT_REQUEST_ID_ATTRIBUTE = "requestId";
+
+    private static Map<String, String> getMappingsV1200() {
+        Map<String, String> mappings = new HashMap<>(4);
+        // messaging mapping, attributes are defined in com.azure.core.amqp.implementation.ClientConstants
+        mappings.put(ENTITY_PATH_KEY, "messaging.destination");
+        mappings.put(HOST_NAME_KEY, "net.peer.name");
+        mappings.put(CLIENT_REQUEST_ID_ATTRIBUTE, "az.client_request_id");
+        mappings.put(SERVICE_REQUEST_ID_ATTRIBUTE, "az.service_request_id");
+
+        return Collections.unmodifiableMap(mappings);
+    }
+
+    public static Attributes convert(Map<String, Object> attributeMap, OpenTelemetrySchemaVersion schemaVersion) {
+        if (attributeMap == null || attributeMap.isEmpty()) {
+            return Attributes.empty();
+        }
+
+        Map<String, String> mappings = getMappingsForVersion(schemaVersion);
+
+        AttributesBuilder builder = Attributes.builder();
+        for (Map.Entry<String, Object> kvp : attributeMap.entrySet()) {
+            Objects.requireNonNull(kvp.getKey(), "'key' cannot be null.");
```

Comment:
Let's make the message a bit clearer in this case to say why this can't be null, maybe:

`The attribute Map being converted to Attributes cannot a null key which isn't allowed in Attributes.`

At least something which guides the user to what is wrong.